### PR TITLE
SIDM-4772 Fix error reporting for setting user password.

### PR DESCRIPTION
**Pull request by radoslaw-orlowski-amido**
### Change description ###

Fix issues with inconsistent error reporting on password creation page.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java [hunk 1]
```diff
@@ -44,36 +44,34 @@ public ValidationService(
...
-            model.put("password1", password1);
-            return false;
-        }
+        final boolean password1HasErrors = !errorForPassword1.isEmpty();
+        final boolean password2HasErrors = !errorForPassword2.isEmpty();
 
-        if (containsIllegalCharacters(password1)) {
-            ErrorHelper.showError("Error", "public.common.error.invalid.password.illegal-characters", "public.common.error.password.details", "", model);
-            model.put("password1", password1);
+        if (password1HasErrors || password2HasErrors) {
```
Comment (tbamido) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java:67]: code coverage related:
test when password1HasErrors
test when only password2HasErrors

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java [hunk 2]
```diff
@@ -44,36 +44,34 @@ public ValidationService(
...
-            ErrorHelper.showError("Error", "public.common.error.password.not.empty", "public.common.error.enter.password", "", model);
-            model.put("password2", password2);
-            return false;
-        }
-        if (StringUtils.isEmpty(password2)) {
-            ErrorHelper.showError("Error", "public.common.error.password.not.empty", "", "public.common.error.enter.password", model);
-            model.put("password1", password1);
-            return false;
+            errorForPassword1 = "public.common.error.enter.password";
+        } else if (password1.length() < passwordMinLength || password1.length() > passwordMaxLength || containsIllegalCharacters(password1)) {
```
Comment (tbamido) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java:53]: 3 different tests here
pass1 too short
pass1 too long
pass1 illegal

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java (hunk 1 above)
Comment (radoslaw-orlowski-amido) [reply src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java:67]: done

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java (hunk 2 above)
Comment (radoslaw-orlowski-amido) [reply src/main/java/uk/gov/hmcts/reform/idam/web/strategic/ValidationService.java:53]: already in the tests

**Review by tbamido — APPROVED** [review]

**Pull request merged**